rotkehlchen/history/events/structures/swap.py

```diff
@@ -0,0 +1,201 @@
+from typing import TYPE_CHECKING, Any, Literal
+
+from rotkehlchen.accounting.mixins.event import AccountingEventType
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.constants import ZERO
+from rotkehlchen.constants.location_details import get_formatted_location_name
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.history.events.structures.base import HistoryBaseEntry, HistoryBaseEntryType
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.history.events.utils import create_event_identifier
+from rotkehlchen.serialization.deserialize import deserialize_fval
+from rotkehlchen.types import Location, TimestampMS
+
+from .base import HISTORY_EVENT_DB_TUPLE_WRITE
+
+if TYPE_CHECKING:
+    from more_itertools import peekable
+
+    from rotkehlchen.accounting.mixins.event import AccountingEventMixin
+    from rotkehlchen.accounting.pot import AccountingPot
+    from rotkehlchen.fval import FVal
+
+
+class SwapEvent(HistoryBaseEntry):
+    """Swap event representing trades on exchanges."""
```

Comment:
Is this only exchanges?

```suggestion
    """Swap event representing trades on exchanges, defi and more."""
```